moveit_core/planning_request_adapter/src/planning_request_adapter.cpp

```diff
@@ -42,22 +42,30 @@ namespace planning_request_adapter
 {
 namespace
 {
-bool callPlannerInterfaceSolve(const planning_interface::PlannerManager& planner,
-                               const planning_scene::PlanningSceneConstPtr& planning_scene,
-                               const planning_interface::MotionPlanRequest& req,
-                               planning_interface::MotionPlanResponse& res)
+moveit::core::MoveItErrorCode callPlannerInterfaceSolve(const planning_interface::PlannerManager& planner,
+                                                        const planning_scene::PlanningSceneConstPtr& planning_scene,
+                                                        const planning_interface::MotionPlanRequest& req,
+                                                        planning_interface::MotionPlanResponse& res)
 {
   planning_interface::PlanningContextPtr context = planner.getPlanningContext(planning_scene, req, res.error_code_);
   if (context)
-    return context->solve(res);
+  {
+    // TODO(andyz): consider returning a moveit::core::MoveItErrorCode from context->solve()
+    bool result = context->solve(res);
+    return result ? moveit::core::MoveItErrorCode::SUCCESS : moveit::core::MoveItErrorCode::FAILURE;
```

Comment:
As long as `context->solve()` doesn't return an actual error code, the distinction between `SUCCESS` and `FAILURE` is the same as `true` and `false`.

Reply:
@rhaschke you are right that the short-circuiting was never implemented. I tried on this branch but failed so far:  https://github.com/AndyZe/moveit2/tree/andyz/adapter_short_circuit

The PR title should have been changed but this way matches with what was merged into MoveIt2. I'm sorry about that.

There is an improvement in the returned code, though, as described here: https://github.com/ros-planning/moveit2/pull/1605#issue-1399761923

Previously, if planning failed, a 0 was always returned. That's not very informative.

Reply:
I have read the original PR description. 
But I couldn't find the source code line generating the output mentioned there, e.g.:
`[moveit_cpp_tutorial-4] [ERROR] [1665063576.588242808] [moveit.planning_request_adapter]: Adapter  result: -10`
Could you please point that out?

I think the same output result could be easily achieved by printing `res.error_code_` instead of the return value - without changing the code as much as here.

Reply:
Yes, i'll make a test branch for you in a bit

Reply:
Here are 2 branches for you. The first one logs adapter success codes prior to this PR. The second logs post-PR.

https://github.com/AndyZe/moveit/tree/andyz/adapter_logging1
https://github.com/AndyZe/moveit/tree/andyz/adapter_logging2

If you check out either branch, add a collision obstacle in RViz, then try to plan a motion, you should see the prints. This is what I get:

**1**
```
[ERROR] [1667508713.025458736]: panda_arm/panda_arm: Motion planning start tree could not be initialized!
[ WARN] [1667508713.025508832]: ParallelPlan::solve(): Unable to find solution by any of the threads in 0.000220 seconds
[ WARN] [1667508713.033321659]: Goal sampling thread never did any work.
[ INFO] [1667508713.033496990]: Unable to solve the planning problem
[ERROR] [1667508713.033546494]: Adapter result: 0
[ERROR] [1667508713.033600887]: Adapter result: 0
[ERROR] [1667508713.033633277]: Adapter result: 0
[ERROR] [1667508713.033645159]: Adapter result: 0
[ERROR] [1667508713.033709058]: Adapter result: 0
[ERROR] [1667508713.033737349]: Adapter result: 0
[ WARN] [1667508713.054276705]: Fail: ABORTED: Catastrophic failure
```

**2**
```
[ INFO] [1667510127.223233904]: Start state appears to be in collision with respect to group panda_arm
[ WARN] [1667510127.315109714]: Unable to find a valid state nearby the start state (using jiggle fraction of 0.050000 and 100 sampling attempts).
[ERROR] [1667510127.315232783]: Adapter result: val: -10

[ERROR] [1667510127.315271245]: Adapter result: val: -10

[ERROR] [1667510127.315306039]: Adapter result: val: -10

[ERROR] [1667510127.315342869]: Adapter result: val: -10

[ERROR] [1667510127.315376119]: Adapter result: val: -10
```

I hope you do find an easier way!

Reply:
Also, it looks like the `master` branch has collision detection issues unrelated to this PR. Sigh

Reply:
> it looks like the `master` branch has collision detection issues unrelated to this PR.

What exactly do you mean? I didn't notice any issues.

Reply:
> What exactly do you mean? I didn't notice any issues.

It doesn't matter right now. Let's focus on this first.